src/scenario/scenario_events_xml.c

```diff
@@ -0,0 +1,530 @@
+    }
+
+    const char *name = xml_parser_get_current_element_name();
+    condition_types type = get_condition_type_from_element_name(name);
+    if (type == CONDITION_TYPE_UNDEFINED) {
+        log_info("Invalid condition type specified", 0, 0);
+        return 0;
+    }
+
+    scenario_condition_t *condition = scenario_event_condition_create(data.current_event, type);
+    return condition_populate_parameters(condition);
+}
+
+static action_types get_action_type_from_element_name(const char *name)
+{
+    for (action_types i = ACTION_TYPE_MIN; i < ACTION_TYPE_MAX; i++) {
+        const char *action_name = scenario_events_parameter_data_get_actions_xml_attributes(i)->xml_attr.name;
+        if (xml_parser_compare_multiple(action_name, name)) {
+            return i;
+        }
+    }
+    return ACTION_TYPE_UNDEFINED;
+}
+
+static int action_populate_parameters(scenario_action_t *action)
+{
+    scenario_action_data_t *data = scenario_events_parameter_data_get_actions_xml_attributes(action->type);
+    int success = 1;
+    success &= xml_import_special_parse_attribute(&data->xml_parm1, &action->parameter1);
+    success &= xml_import_special_parse_attribute(&data->xml_parm2, &action->parameter2);
+    success &= xml_import_special_parse_attribute(&data->xml_parm3, &action->parameter3);
+    success &= xml_import_special_parse_attribute(&data->xml_parm4, &action->parameter4);
+    success &= xml_import_special_parse_attribute(&data->xml_parm5, &action->parameter5);
+
+    return success;
+}
+
+static int xml_import_create_action(void)
+{
+    if (!data.success) {
+        return 0;
+    }
+
+    const char *name = xml_parser_get_current_element_name();
+    action_types type = get_action_type_from_element_name(name);
+    if (type == ACTION_TYPE_UNDEFINED) {
+        xml_import_log_error("Invalid action type specified");
+        log_info("Invalid action type specified", 0, 0);
+        return 0;
+    }
+
+    scenario_action_t *action = scenario_event_action_create(data.current_event, type);
+    return action_populate_parameters(action);
+}
+
+static void xml_import_log_error(const char *msg)
+{
+    data.success = 0;
+    data.error_line_number = xml_parser_get_current_line_number();
+    strcpy(data.error_message, msg);
+    log_error("Error while import scenario events from XML. ", data.error_message, 0);
+    log_error("Line:", 0, data.error_line_number);
+
+    string_copy(translation_for(TR_EDITOR_IMPORT_LINE), data.error_line_number_text, 50);
+    int length = string_length(data.error_line_number_text);
+
+    uint8_t number_as_text[15];
+    string_from_int(number_as_text, data.error_line_number, 0);
+    string_copy(number_as_text, data.error_line_number_text + length, 50);
+
+    window_plain_message_dialog_show_with_extra(
+        TR_EDITOR_UNABLE_TO_LOAD_EVENTS_TITLE, TR_EDITOR_CHECK_LOG_MESSAGE,
+        string_from_ascii(data.error_message),
+        data.error_line_number_text);
+}
+
+#pragma region ATTRIBUTE_PARSING
```

Comment:
I understand why this may be useful, but it throws a warning on older (< 13) versions of GCC.

So it's best removed (both this and `endregion`).

Reply:
Was more useful before I had refactored the functions down to about a third of the initial length. (Don't ask.)
Removed.